internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -1011,3 +976,93 @@ func isLatestBuildPipelineRunInComponent(pipelineRun *tektonv1.PipelineRun, pipe
 	}
 	return true
 }
+
+// haveAllPipelineRunProcessedForPrGroup checks if all build plr has been processed for the given pr group
+func (a *Adapter) haveAllPipelineRunProcessedForPrGroup(prGroup, prGroupHash string) (bool, error) {
+	pipelineRuns, err := a.loader.GetPipelineRunsWithPRGroupHash(a.context, a.client, a.snapshot, prGroupHash)
+	if err != nil {
+		a.logger.Error(err, fmt.Sprintf("Failed to get build pipelineRuns for given pr group hash %s", prGroupHash))
+		return false, err
+	}
+
+	for _, pipelineRun := range *pipelineRuns {
+		pipelineRun := pipelineRun
```

Comment:
why do we need this assignment ?

Reply:
It is the same usage in other loop using `&` to avoid G601, let me comment it.